util/grpc/grpc.go

```diff
@@ -149,3 +149,12 @@ func TestTLS(address string) (*TLSTestResult, error) {
 	}
 	return nil, err
 }
+
+func WithTimeout(duration time.Duration) grpc.UnaryClientInterceptor {
```

Comment:
What about the solution here? https://stackoverflow.com/questions/37414825/specify-a-deadline-with-go-grpc-for-peer-to-peer-connections

Reply:
Looks like it suggests using `context.WithTimeout` https://stackoverflow.com/a/44823585/7898729: 

```
ctx, _ := context.WithTimeout(context.Background(), 100 * time.Millisecond)
r, err := c.SayHello(ctx, &pb.HelloRequest{Name: name})
```

I'm using the same approach here. Interceptor is added to apply timeout everywhere